### PR TITLE
refactor the CURDPrecompiled interface

### PR DESCRIPTION
**Pull request by sulenn**

### CODE REVIEW

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ precompiled/crud/crud_service_test.go  -5
         

Clones removed
==============
+ precompiled/crud/crud_service_test.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5828490&bid=19083624)

File: precompiled/crud/crud_service.go [hunk 1]
```diff
@@ -49,9 +49,8 @@ func NewCRUDService(client *client.Client, privateKey *ecdsa.PrivateKey) (*CRUDS
 	return &CRUDService{crud: crudInstance, tableFactory: tableInstance, crudAuth: auth, client: client}, nil
 }
 
-// CreateTable returns the status of the creating that 0 represents succeed
-func (service *CRUDService) CreateTable(table *Table) (int, error) {
-	tx, err := service.tableFactory.CreateTable(service.crudAuth, table.GetTableName(), table.GetKey(), table.GetValueFields())
+func (service *CRUDService) CreateTable(tableName string, key string, valueField string) (int, error) {
+	tx, err := service.tableFactory.CreateTable(service.crudAuth, tableName, key, valueField)
```
Comment (bxq2011hust) [inline precompiled/crud/crud_service.go:53]:
```suggestion
	tx, err := service.tableFactory.CreateTable(service.crudAuth, tableName, key, valueFields)
```

File: precompiled/crud/crud_service.go [hunk 2]
```diff
@@ -65,8 +64,8 @@ func (service *CRUDService) CreateTable(table *Table) (int, error) {
 }
 
 // Insert entry
-func (service *CRUDService) Insert(table *Table, entry *Entry) (int, error) {
-	if len(table.GetKey()) > TableKeyMaxLength {
+func (service *CRUDService) Insert(tableName string, key string, entry *Entry, optional string) (int, error) {
```
Comment (bxq2011hust) [inline precompiled/crud/crud_service.go:67]:
```suggestion
func (service *CRUDService) Insert(tableName string, key string, entry *Entry) (int, error) {
```

Comment (bxq2011hust) [reply precompiled/crud/crud_service.go:67]: this parameter is not used, just delete it

File: precompiled/crud/crud_service.go [hunk 3]
```diff
@@ -89,8 +88,8 @@ func (service *CRUDService) Insert(table *Table, entry *Entry) (int, error) {
 }
 
 // Update entry
-func (service *CRUDService) Update(table *Table, entry *Entry, condition *Condition) (int, error) {
-	if len(table.GetKey()) > TableKeyMaxLength {
+func (service *CRUDService) Update(tableName string, key string, entry *Entry, condition *Condition, optional string) (int, error) {
```
Comment (bxq2011hust) [inline precompiled/crud/crud_service.go:91]:
```suggestion
func (service *CRUDService) Update(tableName string, key string, entry *Entry, condition *Condition) (int, error) {
```

File: precompiled/crud/crud_service.go [hunk 4]
```diff
@@ -117,16 +116,37 @@ func (service *CRUDService) Update(table *Table, entry *Entry, condition *Condit
 }
 
 // Remove entry
-func (service *CRUDService) Remove(table *Table, condition *Condition) (int, error) {
-	if len(table.GetKey()) > TableKeyMaxLength {
+//func (service *CRUDService) Remove(table *Table, condition *Condition) (int, error) {
```
Comment (bxq2011hust) [inline precompiled/crud/crud_service.go:119]: if not used, what about to delete these

File: precompiled/crud/crud_service.go [hunk 5]
```diff
@@ -140,16 +160,16 @@ func (service *CRUDService) Remove(table *Table, condition *Condition) (int, err
 }
 
 // Select entry
-func (service *CRUDService) Select(table *Table, condition *Condition) ([]map[string]string, error) {
-	if len(table.GetKey()) > TableKeyMaxLength {
+func (service *CRUDService) Select(tableName string, key string, condition *Condition, optional string) ([]map[string]string, error) {
```
Comment (bxq2011hust) [inline precompiled/crud/crud_service.go:163]:
```suggestion
func (service *CRUDService) Select(tableName string, key string, condition *Condition) ([]map[string]string, error) {
```

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ precompiled/crud/crud_service_test.go  -5
         

Clones removed
==============
+ precompiled/crud/crud_service_test.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5828490&bid=19083624)

**Pull request merged**